### ISSUE TITLE
Don't persist path error on startup for forever

### ISSUE THREAD

**Issue by StarpTech** [fastify/fastify-static#14]
When I initialize `fastify-plugin` in folder `/static` and the path does not exists 404 error is thrown. When I create the folder `/static` after that the error still persists and the user doesn't know what's going on. We don't have to persist that error the path is already checked by `send` module and can be handled by the error events.

btw: Isn't possible to use `serve-static` as middleware ?

**Comment by allevo** [comment]
If `/static` doesn't exist, the server start up fails. no?

**Comment by StarpTech** [comment]
No

**Comment by allevo** [comment]
This is a bug.
But there's a test [here](https://github.com/fastify/fastify-static/blob/master/test/static.test.js#L339)

I've written a little example
```js
'use strict'

const fastify = require('fastify')()
const path = require('path')

fastify.register(require('./index'), {
  root: path.join(__dirname, 'unexisting-folder')
})

fastify.listen(0, function (err) {
  console.log(err)
})
```

And this script prints 
```
Error: ENOENT: no such file or directory, stat '/Users/allevo/repos/fastify-static/non-exist-folder'
```

Could you provide some context?

**Comment by allevo** [comment]
**NB:** with `avvio` v3 the registration callback has no needed.

```js
'use strict'

const fastify = require('fastify')()
const path = require('path')

fastify.register(require('./index'), {
  root: path.join(__dirname, 'unexisting-folder')
}, err => {
  console.log(err ? 'error' : '0') // <---- prints 'error'
})

fastify.listen(0, function (err) {
  console.log(err) // <--- err is undefined!
})
```

**Comment by StarpTech** [comment]
Look at example 3 in issue https://github.com/fastify/fastify/issues/391

**Comment by StarpTech** [comment]
Fixed in fastify 0.30.3
